oop1-construct.js

```diff
@@ -1,81 +1,277 @@
-//1. Create a Class called WhyClass.  This class should have one property, and two methods.  THe property is called "purpose", and stores a string.  This string should clearly explain why classes are used in Object Oriented Programming.  The first method is called explain().  This method should print the contents stored in the purpose 'field' of the class.  The second method is called pieces().  This 2nd method should print out the various parts that are essential to building a class.  Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
+//1. Create a Class called WhyClass.  This class should have one property, and two methods.  THe property is called "purpose", and stores a string.
+// This string should clearly explain why classes are used in Object Oriented Programming.  The first method is called explain().  This method should print
+// the contents stored in the purpose 'field' of the class.  The second method is called pieces().  This 2nd method should print out the various parts that
+// are essential to building a class.  Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
 
 //your code here...
+class WhyClass {
+  purpose;
+
+  constructor() {
+    this.purpose =
+      'Classes are used in OOP so you dont have to rewrite things and you can encapsulate multiple forms of data.';
+  }
+
+  explain() {
+    console.log(this.purpose);
+  }
+
+  pieces() {
+    console.log(
+      'The essential parts of building a class are: properties, functions, and a constructor.'
+    );
+  }
+}
 
+const newClass = new WhyClass();
+newClass.explain();
+newClass.pieces();
+
+/*2. You are exploring the rainforests of the Amazon.  You have observed many different types of wildlife on your excursion.  To jog your memory, some 
+different types of wildlife that you observed are found in the following URL : https://www.ietravel.com/blog/amazon-rainforest-animals-beginners-guide-21-species.  Feel free to use other resources online if you wish.
+
+Create a class that can categorize at least 6 of the creatures that you recall observing the Amazon.  Each creature should have 3 unique characteristics, and 
+2 unique actions that it can do. */
+
+class Dolphin {
```

Comment:
a base class or interface would be much better here, so you could create all 6 animals (in a list) and loop through them for the output